spec/common/modules/shared_helper.js

```diff
@@ -11,6 +11,31 @@ define(['spec/common/modules/testapp_module', 'spec/common/modules/client_module
 			/* IANA reserved; requests to it will hang forever */
 			unroutableAddress = 'http://10.255.255.1/';
 
+		var browserBufferCompare = function(buf1, buf2) {
+			if(!buf1) return -1;
+			if(!buf2) return 1;
+			buf1 = clientModule.Ably.Realtime.BufferUtils.toWordArray(buf1);
+			buf2 = clientModule.Ably.Realtime.BufferUtils.toWordArray(buf2);
+			buf1.clamp(); buf2.clamp();
+
+			var cmp = buf1.sigBytes - buf2.sigBytes;
+			if(cmp != 0) return cmp;
+			buf1 = buf1.words; buf2 = buf2.words;
+			for(var i = 0; i < buf1.length; i++) {
+				cmp = buf1[i] - buf2[i];
+				if(cmp != 0) return cmp;
+			}
+			return 0;
+		};
+
+		var nodeBufferCompare = function(buf1, buf2) {
+			if(!buf1) return -1;
+			if(!buf2) return 1;
+			return require('buffertools').compare(buf1, buf2);
```

Comment:
But since this was written we have https://nodejs.org/api/buffer.html#buffer_buf_compare_target_targetstart_targetend_sourcestart_sourceend

for node > 0.11.13